public/app/core/components/RolePicker/utils.ts

```diff
@@ -3,3 +3,19 @@ import { Role } from 'app/types';
 export const isNotDelegatable = (role: Role) => {
   return role.delegatable !== undefined && !role.delegatable;
 };
+
+// addDisplayNameForFixedRole provides a fallback name for fixed roles
+// this is "incase" a fixed role is introduced but without a displayname set
+// example: currently this would give:
+// fixed:datasources:name -> datasources name
+// fixed:datasources:admin      -> datasources admin
+export const addDisplayNameForFixedRole = (role: Role) => {
+  const fixedRolePrefix = 'fixed:';
+  if (role.name.startsWith(fixedRolePrefix) && !role.displayName) {
```

Comment:
```suggestion
  if (!role.displayName && role.name.startsWith(fixedRolePrefix)) {
```
More clear and probably a bit cheaper (`!role.displayName` is cheaper for evaluation and if it's `false`, then the second one won't be evaluated at all).

Reply:
ah, nice improvement